pkg/controller/deployment/deployment_controller.go

```diff
@@ -413,9 +430,56 @@ func (dc *DeploymentController) enqueueDeployment(deployment *extensions.Deploym
 	// periodically relist all deployments there will still be some ReplicaSet instability. One
 	//  way to handle this is by querying the store for all deployments that this deployment
 	// overlaps, as well as all deployments that overlap this deployments, and sorting them.
+
+	// Relist all deployment for overlaps, and avoid syncing the newer overlapping ones (only sync the oldest one)
+	deployments, err := dc.dStore.List()
+	selector, err := unversioned.LabelSelectorAsSelector(deployment.Spec.Selector)
+	if err != nil {
+		glog.Errorf("Deployment %s/%s has invalid label selector: %v", deployment.Namespace, deployment.Name, err)
+		return
+	}
+	overlapping := false
+	for _, d := range deployments {
+		if !selector.Empty() && selector.Matches(labels.Set(d.Spec.Template.Labels)) && d.UID != deployment.UID {
+			overlapping = true
+			dc.markDeploymentOverlap(*deployment, d)
+			dc.markDeploymentOverlap(d, *deployment)
+			// skip syncing this one if older overlapping one is found
+			shouldSkip := d.CreationTimestamp.Before(deployment.CreationTimestamp)
```

Comment:
update selector should still be allowed. simply check creation timestamp is not enough to decide whether we can skip.